.github/workflows/ecosystem-ci-trigger.yml

```diff
@@ -22,57 +66,147 @@ jobs:
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: workflowFileName,
+            })
+            const workflowId = workflow.data.id
+            console.log(`Waiting for workflow ID ${workflowId} to complete...`)
+
+            const maxRetries = 60 // Wait up to 10 minutes
+            const delay = 10000 // 10 seconds
+            let completed = false
+
+            for (let i = 0; i < maxRetries; i++) {
+              const runsData = await github.rest.actions.listWorkflowRuns({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                workflow_id: workflowId,
+                head_sha: prData.commit,
+                per_page: 100,
+                page: 1,
+              })
+
+              const runs = runsData.data.workflow_runs
+
+              if (runs.length > 0) {
+                const latestRun = runs[0]
+                console.log(`Latest run status: ${latestRun.status}, conclusion: ${latestRun.conclusion}`)
+                if (latestRun.status === 'completed') {
+                  if (latestRun.conclusion === 'success') {
+                    console.log('Preview release workflow completed successfully.')
+                    completed = true
+                    break
+                  } else {
+                    throw new Error('Preview Release workflow failed.')
+                  }
+                }
+              }
+
+              console.log(`Retrying... (${i + 1}/${maxRetries})`)
+              await new Promise(resolve => setTimeout(resolve, delay))
             }
-      - id: generate-token
-        uses: tibdex/github-app-token@v2
+
+            if (!completed) {
+              throw new Error('Preview Release workflow did not complete in time.')
+            }
+
+            // Remove the 'rocket' reaction
+            if (reaction) {
+              await github.rest.reactions.deleteForIssueComment({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                comment_id: context.payload.comment.id,
+                reaction_id: reaction,
+              })
+              console.log('Removed "rocket" reaction.')
+            }
+
+      - name: Checkout
+        uses: actions/checkout@v4
         with:
-          app_id: ${{ secrets.ECOSYSTEM_CI_GITHUB_APP_ID }}
-          installation_retrieval_payload: "${{ github.repository_owner }}/vite-ecosystem-ci"
-          private_key: ${{ secrets.ECOSYSTEM_CI_GITHUB_APP_PRIVATE_KEY }}
-      - uses: actions/github-script@v7
+          fetch-depth: 0
+
+      - name: Check Commit Hash Ambiguity
+        id: check_ambiguity
+        run: |
+          HEAD_SHA=${{ steps.get-pr-data.outputs.head_sha }}
+          COMMIT_SHORT=${HEAD_SHA:0:7}
+
+          if git show "$COMMIT_SHORT"; then
+            echo "COLLISION=false" >> $GITHUB_ENV
+          else
+            echo "COLLISION=true" >> $GITHUB_ENV
+          fi
```

Comment:
Let me confirm when these steps can be removed. Is the following understanding right?

It cannot be removed now because even if we passed the full hash to pr.pkg.new, pr.pkg.new will try to match with 7char prefix if it didn't match with the full hash.
It can be removed after all the 7char is purged. How long does it take for that? If the date is known, could we have a comment that says "we can remove this in ..."?

Reply:
> will try to match with 7char prefix

exactly, any release made before yesterday, will match the 7char version of commits (yesterday's release fixes that completely but only gets applied after the new commits)

if we do not use any of the releases made before for a month, it'd get remove! but if we keep using it, the timeline would be expanded to at most 6 months and at that time, it'd be force deleted.

Reply:
done!